SCClassLibrary/Common/Collections/EnvironmentRedirect.sc

```diff
@@ -31,11 +31,20 @@ EnvironmentRedirect {
 		dispatch.value(key, obj);
 	}
 
+	removeAt { arg key;
+		var result = envir.removeAt(key);
+		if (result.notNil) { dispatch.value(key, nil) };
+		^result
+	}
+
+	// if used in a distributed system, these local methods act without
+	// calling the dispatch and thereby redistributing the data recursively
```

Comment:
Are distributed systems where an envir like an Ndef/Proxyspace of one client is in fact controlled by multiple clients? Is this a thing?

Reply:
this was the original use. The implementation is in `JITLibExtensions` (`Public`).